x-pack/solutions/observability/plugins/inventory/public/components/search_bar/entity_types_multi_select.tsx

```diff
@@ -116,7 +117,10 @@ export function EntityTypesMultiSelect() {
             handleEntityTypeChecked(
               newOptions
                 .filter((item) => item.checked)
-                .reduce<EntityType>((acc, curr) => ({ ...acc, [curr.label]: curr.checked! }), {})
+                .reduce<EntityType>(
+                  (acc, curr) => ({ ...acc, [curr.key as string]: curr.checked! }),
```

Comment:
The type cast won't help us in the case when the key is not a string - should we rather convert it to `string` instead (I assume it can't be a falsy value)? 

```suggestion
                  (acc, curr) => ({
                    ...acc,
                    [String(curr.key)]: curr.checked!,
                  }),
```

Reply:
This will actually put more load on the garbage collector and memory allocation. I'd suggest doing this:

```ts
.reduce<EntityType>((acc, curr) => {
  acc[curr.key as string] = curr.checked!;
  return acc;
}, {} as EntityType);
```

Reply:
@jennypavlova keys of objects are automatically stringified, so a type cast suffices (although converting it to String is fine too)

Reply:
thanks for the suggestions and the review, applied @crespocarlos's one

Reply:
@dgieselaar Yeah, that's true, I wanted to find a way to avoid the type-casting in this case ( I first thought of an issue of having "undefined" as a key here and did a check for that beforehand but then I removed that part as it won't be the case here) but what @crespocarlos suggested is a better option to handle the types 👍